src/bin/debugger.rs

```diff
@@ -0,0 +1,35 @@
+use std::{
+    io::{stdin, BufRead, BufReader, Write},
+    net::TcpStream,
+};
+
+use lc3_vm::constants::{LOCAL_HOST, PORT, STREAM_DATA_SEPARATOR};
+
+fn print_instructions() {
+    println!("Instructions: ");
+    println!("<n> to execute one instruction");
+    println!("<a number> to execute that number of instructions (recommended)");
+    println!("<r> to print the state of the registers of the program");
+    println!("remember to use the program you are debugging if it needs user input");
+}
+
+fn main() -> Result<(), String> {
+    let connection_address = format!("{}:{}", LOCAL_HOST, PORT);
+    let stream = TcpStream::connect(connection_address).map_err(|error| error.to_string())?;
```

Comment:
Also, maybe a datagram oriented protocol makes more sense here.

Reply:
Never tried using the IPC unix sockets before and really makes sense to use them in this case for what I read, but why the datagrams ones? It was a nice practice as it was the first time I used `UDP` sockets, but the codes feels way harder to understand (probably the handling of that type of connections that I made wasn't very optimal)

Reply:
I think more in how the message is meant to be interpreted, the debugger is using discrete commands rather than a stream of data from my POV, so you avoid having to keep track of separators and what not. Each message is always a complete command and nothing more, and each message has an associated size.
But you're right about the code seeming harder, it may make sense to revert to stream.

Reply:
I will have that in mind, usually I just go with `TCP` for convenience whenever there are sockets, but knowing the size to avoid separators with `UDP` is interesting